themes-default/slim/src/components/helpers/quality-chooser.vue

```diff
@@ -0,0 +1,349 @@
+                    this.archive = true;
+                } else {
+                    change = 'decrease';
+                }
+                html += 'Backlog will ' + change + ' by <b>' + variation + '</b> episodes.';
+            }
+
+            return {
+                status,
+                html
+            };
+        }
+    },
+    async mounted() {
+        await waitFor(() => this.qualityValues.length > 0, 100, 3000);
+
+        const { isQualityPreset, keep, initialQuality } = this;
+        this.selectedQualityPreset = keep === 'keep' ? 'keep' : (isQualityPreset(initialQuality) ? initialQuality : 0);
+        this.setQualityFromPreset(this.selectedQualityPreset, initialQuality);
+    },
+    methods: {
+        isQualityPreset(quality) {
+            return this.getQualityPreset({ value: quality }) !== undefined;
+        },
+        async archiveEpisodes() {
+            this.archivedStatus = 'Archiving...';
+
+            const url = `series/${this.showSlug}/operation`;
+            const response = await api.post(url, { type: 'ARCHIVE_EPISODES' });
+
+            if (response.status === 201) {
+                this.archivedStatus = 'Successfully archived episodes';
+                // Recalculate backlogged episodes after we archive it
+                this.$asyncComputed.backloggedEpisodes.update();
+            } else if (response.status === 204) {
+                this.archivedStatus = 'No episodes to be archived';
+            }
+            // Restore button text
+            this.archiveButton.text = 'Finished';
+            this.archiveButton.disabled = true;
+        },
+        setQualityFromPreset(preset, oldPreset) {
+            // If empty skip
+            if (preset === undefined || preset === null) {
+                return;
+            }
+
+            // [Mass Edit] If changing to/from `keep`, restore the original value
+            if ([preset, oldPreset].some(val => val === 'keep')) {
+                preset = this.initialQuality;
+            // If preset is custom, set to last preset (provided it's not null)
+            } else if ((preset === 0 || !this.isQualityPreset(preset)) && oldPreset !== null) {
+                preset = oldPreset;
+            }
+
+            const { allowed, preferred } = this.splitQuality(preset);
+            this.allowedQualities = allowed;
+            this.preferredQualities = preferred;
+        }
+    },
+    watch: {
+        // eslint-disable-next-line no-warning-comments
+        /*
+        FIXME: Remove this watch and the `this.lock` hack.
+        This is causing the preset selector to change from `Custom` to a preset,
+        when the correct qualities for that preset are selected.
+        */
+        async overallQuality(newValue) {
+            if (this.lock) {
+                return;
+            }
+
+            // Wait for the store to get populated.
+            await waitFor(() => this.qualityValues.length > 0, 100, 3000);
```

Comment:
> Is the reason your using this, cause you need to wait for a prop and for a computed?

Yes, I need to wait for the store values and for the actual value of overallQuality to be passed down (`editShow` still sends it down "late" because it has to fetch the show info)

When `mounted()` or this watch are called, the `qualityValues` (from the store) isn't populated with values,
and the component doesn't work properly ([has empty custom quality on edit show](https://user-images.githubusercontent.com/10238474/58765793-6696a400-857f-11e9-8c0e-18546b404f84.png)).

On a side note, when more of the app will be using `app.js` we can actually delay the mounting of the Vue app to the DOM until after the config has been fetched. Which I think will help with the issue.

> What would happen if you'd put a watch on both?

I could, and that would work.
But `waitFor` just feels easier to do, plus if the `waitFor` times out (3s in this case), you'll know about it.
So it's easier to debug than a watch that isn't firing?

Reply:
If your absolutely positive the waitFor function does not have any blocking side effects, I'm okay with. Pr looked nice!